contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -132,24 +93,31 @@ contract OptimismPortal {
         uint64 _gasLimit,
         bool _isCreation,
         bytes memory _data
-    ) public payable {
-        // Differentiate between sending to address(0)
-        // and creating a contract
-        if (_isCreation && _to != address(0)) {
-            revert NonZeroCreationTarget();
+    ) public payable metered(_gasLimit) {
+        // Just to be safe, make sure that people specify address(0) as the target when doing
+        // contract creations.
+        // TODO: Do we really need this? Prevents some user error, but adds gas.
```

Comment:
Sure, I'd be OK with removing this check, or even splitting creation out into a separate function and emitting a different event (`CreationDeposited`), though that's probably a bit onerous to change in the go code too.